client.go

```diff
@@ -85,15 +88,29 @@ func NewClient(ctx *Context, cfg *Config, w *Wallet) (*Client, error) {
 	funder := ethchannel.NewETHFunder(cb, common.Address(cfg.AssetHolder.addr))
 	c := client.New(acc, dialer, funder, adjudicator, w.w)
 
+	// Set up persistence
+	db, err := leveldb.LoadDatabase(cfg.DatabasePath)
+	if err != nil {
+		return nil, errors.WithMessage(err, "creating/loading database")
+	}
+	persister, err := keyvalue.NewPersistRestorer(db)
+	if err != nil {
+		return nil, errors.WithMessage(err, "creating PersistRestorer")
+	}
+
+	c.EnablePersistence(persister)
 	go c.Listen(listener)
-	return &Client{cfg: cfg, ethClient: ethClient, client: c, w: w.w, onChain: acc, dialer: dialer}, nil
+	return &Client{cfg: cfg, ethClient: ethClient, client: c, persister: persister, w: w.w, onChain: acc, dialer: dialer}, nil
 }
 
 // Close closes the client and its PersistRestorer to synchronize the database.
 // ref https://pkg.go.dev/perun.network/go-perun/client?tab=doc#Channel.Close
 // ref https://pkg.go.dev/perun.network/go-perun/channel/persistence/keyvalue?tab=doc#PersistRestorer.Close
 func (c *Client) Close() {
 	c.client.Close()
```

Comment:
Now that I'm seeing this, shouldn't we also check for an error of this call?

Reply:
Yea actually we could do that. I thought at first that we can only log it anyway, but is probably better to return it.